tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailboxRepository.scala

```diff
@@ -128,45 +127,42 @@ class TeamMailboxRepositoryImpl @Inject()(mailboxManager: MailboxManager) extend
     val session = createSession(teamMailbox)
     SMono.fromPublisher(exists(teamMailbox))
       .filter(teamMailboxExist => teamMailboxExist)
-      .doOnNext(_ => {
-        addRightForMember(teamMailbox.mailboxPath, user, session)
-        addRightForMember(teamMailbox.inboxPath, user, session)
-        addRightForMember(teamMailbox.sentPath, user, session)
-      })
       .switchIfEmpty(SMono.error(TeamMailboxNotFoundException(teamMailbox)))
+      .flatMap(_ => SFlux.zip3(addRightForMember(teamMailbox.mailboxPath, user, session),
+        addRightForMember(teamMailbox.inboxPath, user, session),
+        addRightForMember(teamMailbox.sentPath, user, session))
+        .`then`()
+        .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER))
```

Comment:
So the rationale behind this is that there is some blocking/heavy computation executed on C* driver event loop?

Reply:
Right now, `StoreRightManager#applyRightsCommand` is blocking (block the reactive method).
`TeamMailboxRepository` invoke the block method => DriverTimeoutException

It will be better when `applyRightsCommand` support the reactive method. Then we don't need duplicate block time.